activerecord/test/cases/tasks/database_tasks_test.rb

```diff
@@ -191,79 +207,128 @@ def test_creates_configurations_with_blank_hosts
       }
-
-      ActiveRecord::Base.stubs(:configurations).returns(@configurations)
-      ActiveRecord::Base.stubs(:establish_connection)
     end
 
     def test_creates_current_environment_database
-      assert_called_with(
-        ActiveRecord::Tasks::DatabaseTasks,
-        :create,
-        ["database" => "test-db"],
-      ) do
-        ActiveRecord::Tasks::DatabaseTasks.create_current(
-          ActiveSupport::StringInquirer.new("test")
-        )
+      with_stubbed_configurations_establish_connection do
+
+        assert_called_with(
+          ActiveRecord::Tasks::DatabaseTasks,
+          :create,
+          ["database" => "test-db"],
+        ) do
+          ActiveRecord::Tasks::DatabaseTasks.create_current(
+            ActiveSupport::StringInquirer.new("test")
+          )
+        end
       end
     end
 
     def test_creates_current_environment_database_with_url
-      assert_called_with(
-        ActiveRecord::Tasks::DatabaseTasks,
-        :create,
-        ["url" => "prod-db-url"],
-      ) do
-        ActiveRecord::Tasks::DatabaseTasks.create_current(
-          ActiveSupport::StringInquirer.new("production")
-        )
+      with_stubbed_configurations_establish_connection do
+
+        assert_called_with(
+          ActiveRecord::Tasks::DatabaseTasks,
+          :create,
+          ["url" => "prod-db-url"],
+        ) do
+          ActiveRecord::Tasks::DatabaseTasks.create_current(
+            ActiveSupport::StringInquirer.new("production")
+          )
+        end
       end
     end
 
     def test_creates_test_and_development_databases_when_env_was_not_specified
-      ActiveRecord::Tasks::DatabaseTasks.expects(:create).
-        with("database" => "dev-db")
-      ActiveRecord::Tasks::DatabaseTasks.expects(:create).
-        with("database" => "test-db")
-
-      ActiveRecord::Tasks::DatabaseTasks.create_current(
-        ActiveSupport::StringInquirer.new("development")
-      )
+      with_stubbed_configurations_establish_connection do
+
+        assert_called_with(
+          ActiveRecord::Tasks::DatabaseTasks,
+          :create,
+          [
+            ["database" => "dev-db"],
+            ["database" => "test-db"]
+          ],
+          returns: true
+        ) do
+          ActiveRecord::Tasks::DatabaseTasks.create_current(
+            ActiveSupport::StringInquirer.new("development")
+          )
+        end
+      end
     end
 
     def test_creates_test_and_development_databases_when_rails_env_is_development
       old_env = ENV["RAILS_ENV"]
       ENV["RAILS_ENV"] = "development"
-      ActiveRecord::Tasks::DatabaseTasks.expects(:create).
-        with("database" => "dev-db")
-      ActiveRecord::Tasks::DatabaseTasks.expects(:create).
-        with("database" => "test-db")
 
-      ActiveRecord::Tasks::DatabaseTasks.create_current(
-        ActiveSupport::StringInquirer.new("development")
-      )
+      with_stubbed_configurations_establish_connection do
+
```

Comment:
✂️

Reply:
Thank you. Fixed. Also found and fixed some more here and in the postgresql task.